inc/Symbol.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SYMBOL_H
+#define SYMBOL_H
+
+#include <iostream>
+
+namespace Calculator {
+
+/**
+ * @brief Represents a symbol in a mathematical expression. Symbols can be any single character from a-z.
+ * 
+ */
+template<typename T>
+class Symbol {
+public:
+    const char symbol;
+private:
+    bool is_bound;
+    T bound_value;
+public:
+    Symbol() = delete;
```

Comment:
add doxygen comment

Reply:
All non-overridden functions should have a doxygen comment in the .h file, update elsewhere as needed